tests/core/vm/test_computation.py

```diff
@@ -0,0 +1,184 @@
+import pytest
+
+from evm.vm import (
+    Computation,
+    Message,
+)
+
+from eth_utils import (
+    to_canonical_address,
+)
+
+from evm.exceptions import (
+    ValidationError,
+)
+
+CANONICAL_ADDRESS_1 = to_canonical_address("0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6")
+CANONICAL_ADDRESS_2 = to_canonical_address("0xcd1722f3947def4cf144679da39c4c32bdc35681")
+
+
+@pytest.fixture
+def message():
+    message = Message(
+        origin=to_canonical_address(CANONICAL_ADDRESS_2),
+        to=CANONICAL_ADDRESS_1,
+        sender=to_canonical_address(CANONICAL_ADDRESS_2),
+        value=100,
+        data=b"0x",
+        code=b"0x",
+        gas=100,
+        gas_price=1,
+    )
+    return message
+
+
+@pytest.fixture
+def computation(message):
+    computation = Computation(
+        vm=1,
+        message=message
+    )
+    return computation
+
+
+@pytest.fixture
+def create_error():
+    def computation_error(computation):
+        computation.gas_meter.gas_remaining = 0
+        with computation:
+            computation.extend_memory(32, 64)
+        return computation
+    return computation_error
+
+
+def test_prepare_child_message(computation):
+    child_message = computation.prepare_child_message(
+        gas=100,
+        to=to_canonical_address(CANONICAL_ADDRESS_2),
+        value=200,
+        data=b"0x",
+        code=b"0x"
+    )
+    assert computation.msg.depth == 0
+    assert child_message.depth == 1
+    assert computation.msg.origin == child_message.origin
+    assert computation.msg.gas_price == child_message.gas_price
+    assert computation.msg.storage_address == child_message.sender
+
+
+def test_extend_memory(computation):
+    with pytest.raises(ValidationError):
+        # Start positions needs to be a uint256
+        computation.extend_memory("1", 1)
+        computation.extend_memory(-1, 1)
```

Comment:
This line (and all the other lines in this block) won't get executed as the first line throws the `ValidationError` and then exits this context.  You need individual `with` statements for each error you want to test.

Also, I prefer to avoid long tests like this which test many different things as they are prone to these types of errors.  Can you see about splitting this up into many smaller tests?